instrumentation/opentelemetry-instrumentation-asyncio/tests/test_asyncio_future_cancellation.py

```diff
@@ -0,0 +1,59 @@
+import asyncio
+from unittest.mock import patch
+
+import pytest
+
+from opentelemetry.instrumentation.asyncio import AsyncioInstrumentor
+from opentelemetry.instrumentation.asyncio.environment_variables import (
+    OTEL_PYTHON_ASYNCIO_FUTURE_TRACE_ENABLED,
+)
+from opentelemetry.test.test_base import TestBase
+from opentelemetry.trace import get_tracer
+
+
+class TestTraceFuture(TestBase):
+    @patch.dict(
+        "os.environ", {OTEL_PYTHON_ASYNCIO_FUTURE_TRACE_ENABLED: "true"}
+    )
+    def setUp(self):
+        super().setUp()
+        self._tracer = get_tracer(
+            __name__,
+        )
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
```

Comment:
Is this needed? I think we can just use `asnycio.run()` in place and the framework will handle loop lifecycles

Reply:
When you run it with asyncio.run(), it converts the future to a coroutine, so you're running the trace_coroutine() logic.

Reply:
See comment below about `async def`. Can you just await the future then? If not I'll take your work for it